src/Test/Utilities/Portable/TestBase.cs

```diff
@@ -147,7 +148,10 @@ public static MetadataReference SystemCoreRef
             {
                 if (s_systemCoreRef == null)
                 {
-                    s_systemCoreRef = AssemblyMetadata.CreateFromImage(TestResources.NetFX.v4_0_30319.System_Core).GetReference(display: "System.Core.v4_0_30319.dll");
+                    Interlocked.CompareExchange(
+                        ref s_systemCoreRef,
+                        AssemblyMetadata.CreateFromImage(TestResources.NetFX.v4_0_30319.System_Core).GetReference(display: "System.Core.v4_0_30319.dll"),
```

Comment:
> Do we care about reference identity of the MetadataReference?

Yes.  See the other comment in this PR.  @agocke calls out that reference equality is used here.